### PR TITLE
Feat/governance voting

### PR DESCRIPTION
**Pull request by nicolasbrugneaux**

### CODE REVIEW

File: src/features/governance/hooks/useVote.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+   */
+  const [voteProposal, voteProposalStatus] = useAsyncCallback(
+    async (proposal: SerializedProposal, vote: VoteType) => {
+      if (!address || !managerContract || !voteContract) {
+        throw new Error('vote called before loading completed');
+      }
+      if (proposal.stage !== ProposalStage.Referendum) {
+        throw new Error('vote called on proposal that is not in Referendum stage');
+      }
+      // TODO: make _voteWeight work?
```
Comment (nicolasbrugneaux) [inline src/features/governance/hooks/useVote.ts:45]: @aaronmgdr

File: src/features/governance/components/Details.tsx [hunk 2]
```diff
@@ -20,19 +21,39 @@ interface Props {
...
+  }, [currentVote, hasVoted, voteProposal, proposal]);
+
+  useEffect(() => {
+    void updateGetHasVoted();
+  }, [updateGetHasVoted, getHasVoted]);
+
   const loaded = Boolean(proposal);
   const fetchError = Boolean(loaded && !proposal?.parsedYAML);
 
+  console.log(proposal);
```
Comment (aaronmgdr) [inline src/features/governance/components/Details.tsx:55]: remove

File: src/features/governance/data/getProposals.ts [hunk 3]
```diff
@@ -57,26 +71,56 @@ const runningProposalStages = new Set([ProposalStage.Queued, ProposalStage.Refer
...
+    }
+    return parsed;
+  } catch {
+    console.warn(`Failed to fetch proposal markdown at url ${proposal.metadata.descriptionURL}`);
+
+    return {
+      cgp: proposal.proposalID,
+      title: `Proposal ${proposal.proposalID}`,
+      dateCreated: proposal.metadata.timestamp,
+      author: '',
```
Comment (aaronmgdr) [inline src/features/governance/data/getProposals.ts:117]: if we arent getting author lets just leave it off the object

File: src/features/governance/hooks/useVote.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        value: '',
+      });
+      await sendTransaction(voteProposalTxObject, txOptions);
+      transactionEvent({
+        action: 'voteProposal',
+        status: 'signed_transaction',
+        value: '',
+      });
+      showVoteToast({ vote, proposalID: proposal.proposalID });
+      await reloadStrategy(address);
```
Comment (aaronmgdr) [inline src/features/governance/hooks/useVote.ts:69]: you dont need this. strategy is about choosing validator

File: src/features/governance/data/getProposals.ts (hunk 3 above)
Comment (nicolasbrugneaux) [reply src/features/governance/data/getProposals.ts:117]: We get it from the github yaml tho, this is just the fallback

File: src/features/governance/hooks/useVote.ts (hunk 1 above)
Comment (aaronmgdr) [reply src/features/governance/hooks/useVote.ts:45]: ?

File: src/features/governance/hooks/useVote.ts [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        proposal.index!,
+        vote === VoteType.yes ? voteWeight : zero,
+        vote === VoteType.no ? voteWeight : zero,
+        vote === VoteType.abstain ? voteWeight : zero
+      );
+      const txOptions = { from: address, gasPrice: suggestedGasPrice };
+      transactionEvent({
+        action: 'voteProposal',
+        status: 'initiated_transaction',
+        value: '',
```
Comment (aaronmgdr) [inline src/features/governance/hooks/useVote.ts:60]: value could be the voteweight

Comment (aaronmgdr) [reply src/features/governance/hooks/useVote.ts:60]: or the YES / NO / ABSTAIN

File: src/features/governance/hooks/useVote.ts [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    [address]
+  );
+
+  // /**
+  //  * TODO: out of scope for now
+  //  */
+  // const [upvote, upvoteStatus] = useAsyncCallback(
+  //   async (proposal: SerializedProposal) => {
+  //     if (!address || !managerContract) {
+  //       throw new Error('vote called before loading completed');
```
Comment (aaronmgdr) [inline src/features/governance/hooks/useVote.ts:92]: i think we should just remove this. i dont see upvoting being a feature

**Pull request merged**